File: BACKEND/BillingService/BillingService/Models/Ordertable.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BillingService.Models
{
    [Table("order_table")]
    public class OrderTable
    {
        [Key]
        [Column("order_id")]
        public int OrderId { get; set; }

        [Column("customer_id")]
        public int CustomerId { get; set; }

        [Column("owner_id")]
        public int OwnerId { get; set; }

        [Column("start_date")]
        public DateTime StartDate { get; set; }

        [Column("end_date")]
        public DateTime EndDate { get; set; }

        [Column("payment_status")]
        public string? PaymentStatus { get; set; }

        [Column("delivery_mode")]
        public string? DeliveryMode { get; set; }

        [Column("owner_item_id")]
        public int OwnerItemId { get; set; }
    }
}

File: BACKEND/CustomerService/src/main/java/com/rentit/addtocart/entities/OrderTable.java
// src/main/java/com/rentit/entity/OrderTable.java

package com.rentit.addtocart.entities; 

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "order_table")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderTable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Integer orderId;
    
    @ManyToOne
    @JoinColumn(name = "customer_id")
    private User customer;
    
    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;
    
    @Column(name = "start_date")
    private LocalDate startDate;
    
    @Column(name = "end_date")
    private LocalDate endDate;
    
    @Column(name = "payment_status")
    private String paymentStatus = "PENDING";
    
//    @Column(name = "delivery_mode")
//    private String deliveryMode = "SELF";
    
    @Enumerated(EnumType.STRING)
    @Column(name = "delivery_mode")
    private DeliveryMode deliveryMode = DeliveryMode.SELF;

    
    @ManyToOne
    @JoinColumn(name = "owner_item_id")
    private OwnerItem ownerItem;
}
